### PR TITLE
Deploy Frontend to Production Firebase

### PR DESCRIPTION
**Pull request by susheelk**
- Perform lint checks on pushes to main
- Add action to deploy frontend to prod on merge to main

## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)

## Implementation description
* Changed job names to match appropriate description
* Duplicated the job that deploys to staging, added deploy to prod on merge to `main`

## What should reviewers focus on?
* Verify project names

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have run docker-compose up and my project compiled
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: .github/workflows/firebase-hosting-merge-prod.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# This file was auto-generated by the Firebase CLI
```
Comment (mahadzaryab1) [inline .github/workflows/firebase-hosting-merge-prod.yml:1]: do we need to add the full file here? why can't we just add the main branch to the `branches` tag in `firebase-hosting-merge.yml`?

**Review by mahadzaryab1 — COMMENTED** [review]
added a comment

File: .github/workflows/firebase-hosting-merge-prod.yml (hunk 1 above)
Comment (susheelk) [reply .github/workflows/firebase-hosting-merge-prod.yml:1]: We're deploying to different environments here and there. Here, we're deploying to prod, there we're deploying to staging

Comment (mahadzaryab1) [reply .github/workflows/firebase-hosting-merge-prod.yml:1]: ah got it

File: .github/workflows/firebase-hosting-merge-prod.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - run: echo "REACT_APP_BACKEND_URL=${{ secrets.REACT_APP_BACKEND_URL_PRODUCTION }}" > .env
+      - run: rm -rf node_modules && yarn install --frozen-lockfile && yarn build
+      - uses: FirebaseExtended/action-hosting-deploy@v0
+        with:
+          repoToken: "${{ secrets.GITHUB_TOKEN }}"
+          firebaseServiceAccount: "${{ secrets.FIREBASE_SERVICE_ACCOUNT_CCBC_PRODUCTION }}"
+          channelId: live
+          projectId: ccbc-95e66
```
Comment (susheelk) [inline .github/workflows/firebase-hosting-merge-prod.yml:29]: whoops just realized this is wrong

**Review by mahadzaryab1 — APPROVED** [review]
lgtm after you resolve your comment!

**Pull request merged**